pandas/_libs/interval.pyx

```diff
@@ -81,26 +81,45 @@ cdef class Interval(IntervalMixin):
 
     Parameters
     ----------
-    left : value
-        Left bound for the interval
-    right : value
-        Right bound for the interval
+    left : orderable object
+        Left bound for the interval.
+    right : orderable object
+        Right bound for the interval.
     closed : {'left', 'right', 'both', 'neither'}, default 'right'
         Whether the interval is closed on the left-side, right-side, both or
-        neither
+        neither.
+    
+    Notes
+    -----
+    You must be able to compare the parameters **left** and **right**,
+    and they must satisfy **left <= right**. 
 
     Examples
     --------
+    The examples below show how you can build Intervals of various types
+    of elements. In particular a numeric interval, a time interval and
+    finally a string interval. Given an Interval you can check whether 
+    an element belongs to it.
+
     >>> iv = pd.Interval(left=0, right=5)
     >>> iv
     Interval(0, 5, closed='right')
     >>> 2.5 in iv
     True
-
+    >>> shifted_iv = iv + 3
+    >>> shifted_iv 
+    Interval(3, 8, closed='right')
+    
     >>> year_2017 = pd.Interval(pd.Timestamp('2017-01-01'),
     ...                         pd.Timestamp('2017-12-31'), closed='both')
     >>> pd.Timestamp('2017-01-01 00:00') in year_2017
     True
+    >>> year_2017.length
+    Timedelta('364 days 00:00:00')
```

Comment:
Might be nice to change `year_2017` to 
```python
year_2017 = pd.Interval(pd.Timestamp('2017-01-01'),
                        pd.Timestamp('2018-01-01'), closed='left')
```
That seems like a more accurate representation of a year, since the current version doesn't include anything beyond `2017-12-31 00:00:00`, e.g. `2017-12-31 12:00:00` isn't included in `year_2017` as currently written.  This change would also result in `year_2017.length` being 365 days, which should be more intuitive for users.